packages/backend/src/indexers/EventIndexer.test.ts

```diff
@@ -0,0 +1,247 @@
+import { assert, Logger } from '@l2beat/backend-tools'
+import { ChainId, EthereumAddress, Hash256, UnixTime } from '@lz/libs'
+import { expect, mockObject } from 'earl'
+import { providers } from 'ethers'
+
+import {
+  BlockchainClient,
+  BlockFromClient,
+} from '../peripherals/clients/BlockchainClient'
+import {
+  BlockNumberRecord,
+  BlockNumberRepository,
+} from '../peripherals/database/BlockNumberRepository'
+import { EventRepository } from '../peripherals/database/EventRepository'
+import { IndexerStateRepository } from '../peripherals/database/IndexerStateRepository'
+import { BlockNumberIndexer } from './BlockNumberIndexer'
+import { EventIndexer } from './EventIndexer'
+
+describe(EventIndexer.name, () => {
+  describe(EventIndexer.prototype.update.name, () => {
+    it('should update from start block', async () => {
+      const startBlock = 15
+      const maxEventRange = 10
+      const chainId = ChainId.ETHEREUM
+      const address = EthereumAddress.random()
+      const topics = [
+        ['event Transfer(address from, address to, uint256 value)'],
+      ]
+      const eventsToWatch = [
+        {
+          address,
+          topics,
+        },
+      ]
+
+      const blockFromClient = mockBlockFromClient(startBlock + maxEventRange)
+      const blockchainClient = mockObject<BlockchainClient>({
+        getBlock: async () => blockFromClient,
+        getLogsBatch: async () => [],
+      })
+      const blockRepo = mockObject<BlockNumberRepository>({
+        findAtOrBefore: async (time) =>
+          time.toNumber() === 100 ? mockBlock(100) : undefined,
+        addMany: async () => 0,
+      })
+      const indexer = new EventIndexer(
+        blockchainClient,
+        blockRepo,
+        mockObject<EventRepository>(),
+        mockObject<IndexerStateRepository>(),
+        startBlock,
+        maxEventRange,
+        chainId,
+        eventsToWatch,
+        mockObject<BlockNumberIndexer>({
+          subscribe: () => undefined,
+        }),
+        Logger.SILENT,
+      )
+
+      await indexer.update(0, 100)
+      expect(blockchainClient.getBlock).toHaveBeenCalledTimes(1)
+      expect(blockchainClient.getBlock).toHaveBeenCalledWith(
+        startBlock + maxEventRange,
+      )
+      expect(blockchainClient.getLogsBatch).toHaveBeenCalledTimes(1)
+      expect(blockchainClient.getLogsBatch).toHaveBeenCalledWith(
+        address,
+        topics,
+        startBlock,
+        startBlock + 10,
```

Comment:
Let's limit magic numbers
```suggestion
        startBlock + maxEventRange,
```

Reply:
[f749fae](https://github.com/l2beat/lz-monitoring/pull/45/commits/f749fae1338885babc9e10cf309ea107fb655626)